src/FerryTempo.js

```diff
@@ -156,12 +158,31 @@ export default {
           'VesselPosition': VesselPositionNum,
         };
 
-        // Set portData.
-        targetRoute['portData'][departingPort].BoatAtDock = AtDock && InService;
-        targetRoute['portData'][arrivingPort].PortArrivalTimeMinus = arrivalTimeEta;
-        targetRoute['portData'][arrivingPort].PortETA = epochEta;
-        targetRoute['portData'][departingPort].PortDepartureDelay = boatDelay;
-        targetRoute['portData'][departingPort].PortStopTimer = timeAtDock;
+        /**
+         * Sometimes two boats on the same route will have the same departingPort and arrivingPort. In this case, we need to be careful setting
+         * the port data. For the departing port, we keep the data if we haven't updated that port already or if the current boat is at dock 
+         * (assuming only one boat can be at dock). For the arrival port, we keep the smaller of the two ETA's provided the current is not zero 
+         * (which means we do not have an ETA set).
+         */
+        let routeDirKey = routeAbbreviation + direction;
+        if (!routeDirCache[routeDirKey]) {
+          targetRoute['portData'][departingPort].BoatAtDock =  AtDock;
+          targetRoute['portData'][departingPort].PortDepartureDelay = boatDelay;
+          targetRoute['portData'][departingPort].PortStopTimer = timeAtDock;
+          targetRoute['portData'][arrivingPort].PortArrivalTimeMinus = arrivalTimeEta;
+          targetRoute['portData'][arrivingPort].PortETA = epochEta;
+          routeDirCache[routeDirKey] = true;
```

Comment:
Do we need logic for clearing the `routeDirCache`?

Reply:
The scope of the cache is within the for loop, so it is destroyed after we process all vessel location records.